### DISCUSSION TITLE
Start framework, emotion css support

### DISCUSSION THREAD

**Discussion by nikolayemrikh** [TanStack/router#2817] [General]
Is it possible to use Emotion with the Start framework?
	•	[Emotion SSR Documentation](https://emotion.sh/docs/ssr)
	•	[TanStack Router Start SSR Basics](https://tanstack.com/router/latest/docs/framework/react/start/ssr#ssr-basics)

The defaultStreamHandler accepts a router and streams a response, but I’m not sure how to integrate Emotion

**Comment by jjhiggz** [comment]
I currently have styled components working (not emotion). Not sure how it actually works under the hood, but I didn't really do anything to hook it up. I just used styled components basically as if it was a client side app, and I'm not 100% sure but it sure feels like I'm getting SSR with it too because it appears to render on the screen super fast.

**Reply by jjhiggz** [reply]
JK I am doing a little something, otherwise I get a hydration error...
in my __root.tsx
```ts
import { StyleSheetManager } from "styled-components";

// other stuff...

function RootDocument({ children }: { children: React.ReactNode }) {
  return (
    <html>
      <head>
        <Meta />
      </head>
      <body>
        <StyleSheetManager shouldForwardProp={shouldForwardProp}> {/* <-- this */}
          <Toaster />
          {children}
          <ScrollRestoration />
          <TanStackRouterDevtools position="bottom-right" />
          <Scripts />
        </StyleSheetManager>
      </body>
    </html>
  );
}
```
